feat: adds option to search only for untagged assets

SkwalExe opened the pull request:
## Description

The possibility to search for untagged assets makes organisation easier. This feature allows the user to start tagging their assets exhaustively. Personnally, tagging features are the only thing I'm waiting before switching to Immich entirely!

## How Has This Been Tested?

This adds very little change to the codebase, and similar search parameters, like "Not in any album" aren't tested, so it doesn't look like a necessity here.

<details><summary><h2>Screenshots (if appropriate)</h2></summary>

<!-- Images go below this line. -->

![image](https://github.com/user-attachments/assets/f8d520ac-91a4-424a-9f0f-bfd237a49d14)

</details>


## API Changes
- Added a `isUntagged: bool` parameter to search-related DTOs.


## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation if applicable
- [x] I have no unrelated changes in the PR.
- [x] I have confirmed that any new dependencies are strictly necessary.
- [ ] I have written tests for new code (if applicable)
- [x] I have followed naming conventions/patterns in the surrounding code
- [x] All code in `src/services/` uses repositories implementations for database calls, filesystem operations, etc.
- [x] All code in `src/repositories/` is pretty basic/simple and does not have any immich specific logic (that belongs in `src/services/`)

Commit:
Translation

Commit:
Add untagged search parameter on the server side

Commit:
Add Untagged search option on the web client

danieldietzler reviewed (commented):
Thanks! I wonder if this should be its own field or whether the existing `tagIds` field should be used for it. Like either `tagIds: null` or `tagIds: []`. I feel like the second option is less descriptive but having it nullable could work? Thoughts?

SkwalExe commented:
> Thanks! I wonder if this should be its own field or whether the existing `tagIds` field should be used for it. Like either `tagIds: null` or `tagIds: []`. I feel like the second option is less descriptive but having it nullable could work? Thoughts?

I think null could work, I tried to follow the same logic as `isNotInAlbum` to maintain consistency

danieldietzler commented:
> > Thanks! I wonder if this should be its own field or whether the existing `tagIds` field should be used for it. Like either `tagIds: null` or `tagIds: []`. I feel like the second option is less descriptive but having it nullable could work? Thoughts?
> 
> I think null could work, I tried to follow the same logic as `isNotInAlbum` to maintain consistency

Yeah that makes sense too. @mertalev thoughts?

mertalev commented:
I think null makes sense. It makes for a nicer API if you can't input conflicting options.

Commit:
Improve Untagged filter logic server side

danieldietzler commented:
> > I think null makes sense. It makes for a nicer API if you can't input conflicting options.
> 
> Then let's go with null! Should I implement this logic with `isNotInAlbum` as well?

I think so. If you do, that should be a separate PR though. Thanks :)

Commit:
Small fix (server)

Commit:
let combobox be disabled

Commit:
web: update untagged filter to match new API implementation

SkwalExe commented:
I removed the `isUntagged` option and used `tagIds` instead. In addition, I moved the checkbox under the tags section, for coherence, and made the `search tags` field disabled when needed.

![image](https://github.com/user-attachments/assets/d58a09d1-550d-4ee8-a983-b4ff61b6606f)

danieldietzler reviewed (approved):
Code LGTM, thanks!

Commit:
Fix eslint error

SkwalExe committed:
Merge branch 'main' into main
